tests/foreman/cli/test_errata.py

```diff
@@ -459,6 +460,35 @@ def test_negative_install_by_hc_name_without_org_info(self):
             )
         self.assertIn('Error: Could not find organization', context.exception.stderr)
 
+    @upgrade
+    def test_positive_list_affected_chosts(self):
+        """View a list of affected content hosts for an erratum
+
+        :id: 3b592253-52c0-4165-9a48-ba55287e9ee9
+
+        :Setup: Errata synced on satellite server.
+
+        :Steps: host list --search "applicable_errata = <erratum_id>"
+            --organization-id=<org_id>
+
+        :expectedresults: List of affected content hosts for an erratum is
+            displayed.
+
+        :CaseAutomation: automated
+
+        """
+        with ssh.get_connection() as connection:
+            result = connection.run(
```

Comment:
```suggestion
            result = Host.list({'search': 'applicable_errata = {}'.format(FAKE_2_ERRATA_ID), 'organization-id': org['id'], 'fields': 'Name'})
```
@swadeley  We can change it to something like this.

Reply:
**Can** is not the same as **should**. 
The command is not used elsewhere, so we cannot argue for consistency.
Will the test be faster? Will it be easier to understand or maintain? 

I have another test coming which has multiple commands like this, if I use the Host.list will it set up an SSH connection every time? Using a connection manager is supposed to save time, and all though this test only has the one, related tests will have more than one. So consistency will then become a factor.

Reply:
I would agree with Jameer here. We have abstracted hammer for a reason, and we should use these abstractions when possible. The idea is, it **will** make tests more maintainable, as fixes might be introduced in base classes, instead of hunting all references to command in entire codebase.

These abstractions add `-u {user} -p {pass}`, to ensure hammer calls are properly authenticated. Your command is missing that.

It seems to me your main concern is speed. Let's remember that "premature optimization is the root of all evil." Even in this next test, where you will have more commands, I'd be curious about performance comparison between this approach and using abstractions. I'd like to see numbers, not what seems or should be faster.

Also, I believe @JacobCallahan has some idea for shared SSH session. Of course it will be done in CLI abstractions, so if you use them, your code will benefit from them.

Reply:
> I would agree with Jameer here. We have abstracted hammer for a reason, and we should use these abstractions when possible. The idea is, it **will** make tests more maintainable, as fixes might be introduced in base classes, instead of hunting all references to command in entire codebase.

OK, will change.
> 

> 
> Also, I believe @JacobCallahan has some idea for shared SSH session. Of course it will be done in CLI abstractions, so if you use them, your code will benefit from them.

I remember his e-mail:
> [1] - https://parallel-ssh.org/post/ssh2-python/